llvm/cmake/modules/Findzstd.cmake

```diff
@@ -29,11 +29,11 @@ find_package_handle_standard_args(
 )
 
 if(zstd_FOUND)
-  if(zstd_LIBRARY MATCHES "${zstd_STATIC_LIBRARY_SUFFIX}$")
+  if(zstd_LIBRARY MATCHES "${zstd_STATIC_LIBRARY_SUFFIX}$" AND NOT zstd_LIBRARY MATCHES "\\.dll\\.a$")
```

Comment:
Is there any cmake variable that contains the `.dll.a` suffix as a string? But perhaps we don't want to use that here; for MSVC that (`.lib`) would be equal to the static library suffix, so it would negate the effect of the first part of the condition.

Reply:
I don't know of any such variable, though we're more interested here in how zstd was compiled, while this check still incorrectly assumes the conventions are based on how LLVM will be compiled. I wasn't sure it is even possible to correctly identify the kind of file this is just using pattern matching on the name, but this change is trying to just make it behave more consistently. (When compiling MSVC, this tests against "_static.lib", so adding "not .dll.a" has no impact)

Reply:
Oh, I missed that this used `zstd_STATIC_LIBRARY_SUFFIX`, I thought it was the usual `CMAKE_STATIC_LIBRARY_SUFFIX`.

If it would be the cmake general ones, and if we'd use the variable for import library suffix, like `if (zstd_LIBRARY MATCHES "${CMAKE_STATIC_LIBRARY_SUFFIX}$" AND NOT zstd_LIBRARY MATCHES "${CMAKE_IMPORT_LIBRARY_SUFFIX}$")`, it'd essentially be `if (zstd_LIBRARY MATCHES ".lib$" AND  NOT zstd_LIBRARY MATCHES ".lib$")`.

But this wasn't the case, so nevermind - sorry for the detour.

Reply:
Upstream, the variable used in similar situations appears to be a hard coded list of ".dll.a" combined with CMAKE_FIND_LIBRARY_SUFFIXES:
https://github.com/Kitware/CMake/blob/ede59aac3f28a0ecb535f13bdcb81fbd5c109983/Modules/FindZLIB.cmake#L162-L171